### PR TITLE
Detect markup inside links

### LINKED ISSUE #1243: Allow inspecting current file name
Perhaps by adding `"current_file"` as a possible key for `roxy_meta_get()`, as suggested on Slack by @gaborcsardi.

This could be used in inline R code to insert the current file name, a bit like CPP macros.

### PR DESCRIPTION
**Pull request by lionel-**
Branched from #1260.

Non-text nodes in links can't be parsed to HTML: https://github.com/wch/r-source/blob/eb6d29616/src/library/tools/R/Rd2HTML.R#L31-L32

Currently ``[foo `bar`][x]`` generates invalid Rd that isn't picked up by R CMD check and causes runtime error when opening help pages. The patch detects these cases and throws an informative error.

The error informs about the current file. Adds `roxy_meta` options `current_file` and `current_path`, which closes #1243.

### CODE REVIEW

**Comment by hadley** [comment]
Can you please merge/rebase this?

**Comment by lionel-** [comment]
Now rebased.

File: R/block.R [hunk 1]
```diff
@@ -212,11 +212,22 @@ parse_tags <- function(tokens) {
 
   out <- vector("list", length(tokens))
   for (i in seq_along(tokens)) {
+    roxy_meta_local(current_lines = token_lines(tokens, i))
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/block.R:215]: Oh I just noticed that `markdown()` already has something to do this: https://github.com/r-lib/roxygen2/blob/main/R/markdown.R#L5-L14

So I think you can just error and the wrapper will take care of adding file/line info.

File: R/markdown-link.R [hunk 2]
```diff
@@ -102,6 +102,21 @@ parse_link <- function(destination, contents, state) {
     local_bindings(.env = state, in_link_code = TRUE)
   }
 
+  if (!all(xml_name(contents) %in% c("text", "softbreak", "linebreak"))) {
+    incorrect <- setdiff(unique(xml_name(contents)), c("text", "softbreak", "linebreak"))
+
+    file <- roxy_meta_get("current_file")
+    lines <- roxy_meta_get("current_lines")
+
+    cli::cli_abort(c(
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/markdown-link.R:111]: Even simpler — `state$tag` contains the current tag, so just pass that to `roxy_tag_warning()`.

**Pull request merged**